packages/strapi-plugin-upload/admin/src/containers/HomePage/index.js

```diff
@@ -17,17 +23,75 @@ import AddFilterCTA from '../../components/AddFilterCTA';
 const HomePage = () => {
   const { formatMessage } = useGlobalContext();
   const [reducerState, dispatch] = useReducer(reducer, initialState, init);
-  const [isOpen, setIsOpen] = useState(true);
-  const { data, dataToDelete, _q } = reducerState.toJS();
+  const [isOpen, setIsOpen] = useState(false);
+  const { push } = useHistory();
+  const { search } = useLocation();
+  const query = useQuery();
+  const { data, dataToDelete, _sort, _q } = reducerState.toJS();
   const pluginName = formatMessage({ id: getTrad('plugin.name') });
 
-  const handleClickToggleModal = () => {
-    setIsOpen(prev => !prev);
+  useEffect(() => {
+    const searchParams = getSearchParams();
+
+    Object.keys(searchParams).map(key => {
+      return dispatch({
+        type: 'ON_QUERY_CHANGE',
+        key,
+        value: searchParams[key],
+      });
+    });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [search]);
+
+  const getSearchParams = () => {
+    const params = {};
+    query.forEach((value, key) => {
+      params[key] = value;
+    });
+
+    return params;
   };
-  const handleClearSearch = () => {
+
+  const getUpdatedSearchParams = updatedParams => {
+    return {
+      ...getSearchParams(),
+      ...updatedParams,
+    };
+  };
+
+  const handleChangeParams = ({ key, value }) => {
+    const updatedSearch = getUpdatedSearchParams({ [key]: value });
+    const newSearch = generateSearchFromFilters(updatedSearch);
+    push({ search: newSearch });
+  };
+
+  const handleChangeSearch = ({ target: { value } }) => {
```

Comment:
You should use a single function to handle your input change like in a normal form.